crates/core_arch/src/x86/avx2.rs

```diff
@@ -3743,9 +3743,9 @@ pub unsafe fn _mm256_xor_si256(a: __m256i, b: __m256i) -> __m256i {
 // This intrinsic has no corresponding instruction.
 #[rustc_args_required_const(1)]
 #[stable(feature = "simd_x86", since = "1.27.0")]
-pub unsafe fn _mm256_extract_epi8(a: __m256i, imm8: i32) -> i8 {
+pub unsafe fn _mm256_extract_epi8(a: __m256i, imm8: i32) -> i32 {
```

Comment:
This and the change below is technically a breaking change in the public API. However we are fixing the function here to match the definition. Thoughts?

Reply:
I think that we need a crater run at the very least for this. cc @rust-lang/libs

Reply:
I think this is probably going to break anyone that uses `_mm256_extract_epi8`, no?

It seems like this is probably an okay "wontfix" bug, since I think `i8` is big enough to hold all possible bit patterns returned here.

Reply:
I expect most users of these intrinsics to cast the returned value immediately, which would hide the issue.